include/zephyr/bluetooth/ead.h

```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+
+/** Additional Authenticated Data size in bytes */
+#define BT_EAD_AAD_SIZE	       1
+/** Randomizer size in bytes */
+#define BT_EAD_RANDOMIZER_SIZE 5
+/** nonce size in bytes */
+#define BT_EAD_NONCE_SIZE      13
+/** Key size in bytes */
+#define BT_EAD_KEY_SIZE	       16
+/** Initialisation Vector size in bytes */
+#define BT_EAD_IV_SIZE	       8
+/** MIC size in bytes */
+#define BT_EAD_MIC_SIZE	       4
+
+/**
+ * @brief Encrypt and authenticate the given advertising data. The resulting
+ * data in @p encrypted_ad_data will look like that:
+ *
+ *     - Randomizer is added in the 5 first bytes;
+ *     - AAD is added in the next byte;
+ *     - Encrypted payload is added (payload_size bytes);
+ *     - MIC is added in the last 4 bytes.
+ *
+ * @note The function must be called each time the RPA is updated or the data
+ * are modified.
+ *
+ * @note The term `advertising structure` is used to describe the advertising
+ *       data with the advertising type and the length of those two.
+ *
+ * @param[in]  session_key Key of 16 bytes used for the encryption.
+ * @param[in]  iv Initialisation Vector used to generate the nonce. It must be
+ *             changed each time the Session Key change.
+ * @param[in]  payload Advertising Data to encrypt. Can be multiple advertising
+ *             structure concatenated.
+ * @param[in]  payload_size Size of the Advertising Data to encrypt.
+ * @param[out] encrypted_ad_data Encrypted Adv Data including the Randomizer and
+ *             the MIC. Size must be `BT_EAD_RANDOMIZER_SIZE + BT_EAD_AAD_SIZE +
+ *             payload_size + BT_EAD_MIC_SIZE`.
+ * @retval int Size of @p encrypted_ad_data in bytes if successful, else return
+ *             `-EIO` if an error occurred during the encryption.
+ */
+int bt_ead_encrypt(const uint8_t *session_key, const uint8_t *iv, const uint8_t *payload,
+		   int payload_size, uint8_t *encrypted_ad_data);
```

Comment:
```suggestion
		   size_t payload_size, uint8_t *encrypted_ad_data);
```

Reply:
Why use `size_t`? The maximum size of the advertising data is 1650.

Reply:
Two reasons:
1) It should be a unsigned type (we don't want negative payload sizes)
2) We use `size_t` everywhere else in the BT API to denote array/data lengths, except where it has a maximum size of 254 (e.g. `struct bt_data`)